onnxruntime/core/framework/utils.cc

```diff
@@ -98,6 +98,26 @@ void DefaultFree(void* p) {
 #endif
 }
 
+void ConstructStrings(void* p_data, int64_t elements) {
+  auto* ptr = static_cast<std::string*>(p_data);
+  for (int64_t i = 0; i < elements; ++i) {
+    new (ptr + i) std::string();
+  }
+}
+
+
+void ReleaseTensorBuffer(const AllocatorPtr& allocator, bool is_string, void* p_data, int64_t elements) {
```

Comment:
nit: could be simply called ReleaseBuffer as there's nothing tensor specific about it in the interface. If there is, should be part of the Tensor class.